File: haskell/hanoi_tower.hs
data Tower = A | B | C deriving Show

hanoi :: Int -> Tower -> Tower -> Tower -> Int
hanoi 1 _ _ _ = 1
hanoi n x y z = 
  (hanoi (n - 1) x z y) + 1 + (hanoi (n - 1) y x z)

main = do
  putStrLn $ "Moves: " ++ show (hanoi 32 A B C)

File: clojure/hanoi_tower.clj
(defn hanoi-tower [n a b c]
  (if (= n 1)
    1
    (+ (hanoi-tower (- n 1) a c b) 1 (hanoi-tower (- n 1) b a c))))

(hanoi-tower 32 "a" "b" "c")